test/global.test.ts

```diff
@@ -26,8 +39,39 @@ suiteSetup(async function (this: Mocha.Context): Promise<void> {
         await extension.activate();
     }
 
+    await fse.ensureDir(testFolderPath);
+    await fse.emptyDir(testFolderPath);
+    testWorkspacePath = await initTestWorkspacePath();
+
     ext.outputChannel = new TestOutputChannel();
     ext.ui = testUserInput;
 
     longRunningTestsEnabled = !/^(false|0)?$/i.test(process.env.ENABLE_LONG_RUNNING_TESTS || '');
+
+    // set the user name and email since this seems to be required for everything
+    await cpUtils.executeCommand(undefined, undefined, 'git', 'config', '--global', 'user.name', 'Automated Tester');
```

Comment:
This could definitely mess with someone's git config if they're running tests locally. Also I wonder if the CI machines have their own git config and this could cause problems there, too

Reply:
Hmm, as far as I know, the CI machines don't have these configs set because that was one of the errors I was running into.  But good point about messing with someone running the tests locally.

Would it make sense to check if these configurations exist and only execute if we find that it doesn't?

Reply:
Yes that might make sense. Can show me the exact error you were hitting?

Reply:
https://stackoverflow.com/questions/9617336/how-to-resolve-git-did-not-exit-cleanly-exit-code-128-error-on-tortoisegit

The second answer refers to the error that I was running into.

Reply:
...I mean yes I know how you fixed it, but that link doesn't really show the error. I was hoping you'd link me to a dev ops build log with the error

If I can see the error, I can maybe help find an alternate solution. Regardless, one idea is maybe you can set the config using "--local" instead of "--global"? Another idea is I think you can use [environment variables](https://git-scm.com/docs/git-commit#_commit_information) so maybe you can set those just for this test process

Reply:
Oh, you can see it in this build.  Test 2 has the error: https://dev.azure.com/ms-azuretools/AzCode/_build/results?buildId=34481&view=logs&jobId=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=6e60e40e-3d3d-5182-ba10-afdf8e87d3dd

Though note that error is actually coming from us. https://github.com/microsoft/vscode-azurestaticwebapps/blob/main/src/errors.ts#L57-L59

Reply:
And I think the environment variables is a great idea-- I'll give that a try.